packages/mobile/src/invite/JoinCelo.tsx

```diff
@@ -0,0 +1,308 @@
+    const { cachedName, cachedNumber, cachedCountryCode } = this.props
+
+    if (cachedName === name && cachedNumber === e164Number && cachedCountryCode === countryCode) {
+      this.nextScreen()
+      return
+    }
+
+    if (!name || !e164Number || !isValidNumber || !countryCode) {
+      this.props.showError(ErrorMessages.INVALID_PHONE_NUMBER)
+      return
+    }
+
+    this.setState({ isSubmitting: true })
+    this.props.hideAlert()
+    this.props.setPhoneNumber(e164Number, countryCode)
+    this.props.setName(name)
+    this.props.redeemComplete(false)
+    this.setState({ isSubmitting: false })
+    this.nextScreen()
+  }
+
+  focusOnCode = () => {
+    if (this.codeInput.current) {
+      this.codeInput.current.focus()
+    }
+  }
+
+  scrollToEnd = () => {
+    setTimeout(() => {
+      if (this.scrollView && this.scrollView.current) {
+        this.scrollView.current.scrollToEnd()
+      }
+    }, 1000) // This timeout must long enough or it doesnt not work
+  }
+
+  render() {
+    const { t, language } = this.props
+    const { name } = this.state
+
+    return (
+      <View style={styles.container}>
+        <DevSkipButton nextScreen={Screens.Pincode} />
+        <ScrollView
+          contentContainerStyle={styles.scrollContainer}
+          keyboardShouldPersistTaps="always"
+          ref={this.scrollView}
+        >
+          <View>
+            <NuxLogo style={styles.logo} />
+            <Text style={[fontStyles.h1, styles.center]} testID="InviteWallTitle">
+              {' '}
+              {t('welcomeCelo')}
+            </Text>
+            <Text style={[fontStyles.bodySmall, styles.body]}>{t('joinText.0')}</Text>
+            <Text style={[fontStyles.bodySmall, styles.body]}>
+              {t('joinText.1')}
+              <Text onPress={goToCeloSite} style={fontStyles.link}>
+                {t('joinText.2')}
+              </Text>
+            </Text>
+            <TextInput
+              onFocus={this.scrollToEnd}
+              onChangeText={this.setName}
+              value={name}
+              style={styles.box}
+              placeholderTextColor={colors.inactive}
+              underlineColorAndroid="transparent"
+              enablesReturnKeyAutomatically={true}
+              onSubmitEditing={this.focusOnCode}
+              placeholder={t('fullName')}
+              testID={'NameEntry'}
+            />
+            <PhoneNumberInput
+              setE164Number={this.setE164Number}
+              setCountryCode={this.setCountryCode}
+              setIsValidNumber={this.setIsValidNumber}
+              onInputFocus={this.scrollToEnd}
+              onInputChange={this.onChangePhoneInput}
+              inputCountryPlaceholder={t('chooseCountryCode')}
+              inputPhonePlaceholder={t('phoneNumber')}
+              callingCode={true}
+              lng={language}
+            />
+          </View>
+        </ScrollView>
+        <GethAwareButton
```

Comment:
I don't believe this screen should use a GethAwareButton. 
We use that when we want to prevent clicking until the syncing is done.